### ISSUE TITLE
[BUG]: Unecessary re-render

### ISSUE THREAD

**Issue by allangomessl** [immerjs/use-immer#21]
```js
  const [values, setResources] = useImmer({ })

  useEffect(() => {
    (async () => {
      const { data } = await api.get("v2/api-docs")
      setResources(() => data.paths)
    })();
  }, [setResources])
```

this cause many re-renders because `setResource` always a new instance

---

I believe we must add `useCallback` on line below:
https://github.com/immerjs/use-immer/blob/7e3bc90e2d6088c2d49577c5ea3681640ea67f57/src/index.ts#L16

---

the final code would be like this:
```js
return [val, useCallback(newValue =>
    updateValue(produce(newValue))
  , [updateValue])
```

**Comment by mweststrate** [comment]
I think this one is solved by #22 in 0.3.1

**Comment by yezhi780625** [comment]
> I think this one is solved by #22 in 0.3.1

Hello @mweststrate , It seems not work.
https://codesandbox.io/s/2wv8xw6w7y

when I changed person's name, it trigger `useEffect` every time.

The second return value `updater` function should be wrapped by `useCallback`.

https://github.com/immerjs/use-immer/blob/7e3bc90e2d6088c2d49577c5ea3681640ea67f57/src/index.ts#L16.

**Comment by mweststrate** [comment]
Thanks for the PR! Released as 0.3.2
